File: suitable/SetLed/wrong_code/SetLed.py
# 初始模式，0 表示控制权由开发板管理
mode = 0 
'''
# 模拟写入 GPIO 引脚的值
def write_pin(state):
    if state:
        print("LED is ON")
    else:
        print("LED is OFF")
'''
# 控制 LED 的方法
#为了方便规约，on,off,con分别表示打开，关闭，控制权返还给开发板,对应0,1,2
def set_led(msg,state=True):
    global mode
    #ON-0,OFF-1,CON-2
    msg = int(msg)
    if msg == 0:
        mode = 1
        #write_pin(True)  # 模拟打开 LED
        #print(f"{msg} turn on led")
        state  = True
    elif msg == 1:
        mode = 1
        #write_pin(False)  # 模拟关闭 LED
        #print(f"{msg} turn off led")
        state = False
    elif msg == 2:
        mode = 0  # 控制权返还给开发板
        #wrong:缺少state = False
        #print("Control returned to the board.")
    print(f"msg:{msg} mode:{mode},state:{state}")
    return mode,state

# 测试该方法
#set_led("ON")
#set_led("OFF")
#set_led("CON")
if __name__ == "__main__":
    
    set_led(input("请输入指令："))

File: suitable/SetLed/wrong_code/SetLed.java
import java.util.Scanner;

public class SetLed {
    // 初始模式，0 表示控制权由开发板管理
    private static int mode = 0;

    // 定义一个内部类用于返回多个值
    public static class Result {
        int mode;
        boolean state;

        public Result(int mode, boolean state) {
            this.mode = mode;
            this.state = state;
        }

        @Override
        public String toString() {
            return "mode: " + mode + ", state: " + state;
        }
    }

    // 控制 LED 的方法
    // on=0, off=1, con=2 表示打开、关闭和控制权返还给开发板
    public static Result setLED(int msg) {
        boolean state = true;
        
        // 根据不同的 msg 值设置 mode 和 state
        if (msg == 0) {
            mode = 0;//wrong code
            state = true;
        } else if (msg == 1) {
            mode = 1;
            state = false;
        } else if (msg == 2) {
            mode = 0;  // 控制权返还给开发板
            state = false;
        } else {
            System.out.println("Invalid command");
            return new Result(mode, false);  // 返回当前 mode 和无效的 state
        }

        // 返回 mode 和 state 的结果
        return new Result(mode, state);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("请输入指令 (0: 打开, 1: 关闭, 2: 返还控制权): ");
        int msg = scanner.nextInt();
        Result result = setLED(msg);  // 获取返回的结果
        System.out.println(result);    // 打印结果
        scanner.close();
    }
}
